File: Examples/Chart/WebForm/VB.NET/Examples/WebControl/Printing/NPrintingUC.ascx.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Data
Imports System.Drawing
Imports System.Web
Imports System.Web.UI.WebControls
Imports System.Web.UI.HtmlControls
Imports System.Web.UI

Imports Nevron.GraphicsCore
Imports Nevron.Chart
Imports Nevron.Chart.WebForm

Namespace Nevron.Examples.Chart.WebForm
	Public Partial Class NPrintingUC
		Inherits NExampleUC
		Protected Sub Page_Load(ByVal sender As Object, ByVal e As EventArgs)
			nChartControl1.StateManager = New NChartBatonSessionStateManager(Context, "Nevron.Examples.Chart.WebForm.NPrintingUC")
			nChartControl1.BackgroundStyle.FrameStyle.Visible = False

			If nChartControl1.RequiresInitialization Then
				' enable jittering (full scene antialiasing)
				nChartControl1.Settings.JitterMode = JitterMode.Enabled

				' set a chart title
				Dim title As NLabel = nChartControl1.Labels.AddHeader("Welcome to Nevron Chart for .NET")
				title.TextStyle.BackplaneStyle.Visible = False
				title.TextStyle.FillStyle = New NColorFillStyle(Color.Navy)
				title.TextStyle.ShadowStyle.Type = ShadowType.GaussianBlur
				title.ContentAlignment = ContentAlignment.BottomRight
				title.Location = New NPointL(New NLength(2, NRelativeUnit.ParentPercentage), New NLength(2, NRelativeUnit.ParentPercentage))

				' setup the legend
				Dim legend As NLegend = nChartControl1.Legends(0)
				legend.FillStyle.SetTransparencyPercent(50)
				legend.OuterBottomBorderStyle.Width = New NLength(0, NGraphicsUnit.Pixel)
				legend.OuterLeftBorderStyle.Width = New NLength(0, NGraphicsUnit.Pixel)
				legend.OuterRightBorderStyle.Width = New NLength(0, NGraphicsUnit.Pixel)
				legend.OuterTopBorderStyle.Width = New NLength(0, NGraphicsUnit.Pixel)
				legend.HorizontalBorderStyle.Width = New NLength(0, NGraphicsUnit.Pixel)
				legend.VerticalBorderStyle.Width = New NLength(0, NGraphicsUnit.Pixel)

				' setup the chart
				Dim chart As NChart = nChartControl1.Charts(0)
				chart.Enable3D = True
				chart.Axis(StandardAxis.Depth).Visible = False
				chart.LightModel.SetPredefinedLightModel(PredefinedLightModel.GlitterLeft)
				chart.Projection.SetPredefinedProjection(PredefinedProjection.OrthogonalHalf)
				chart.BoundsMode = BoundsMode.Stretch
				chart.Location = New NPointL(New NLength(10, NRelativeUnit.ParentPercentage), New NLength(17, NRelativeUnit.ParentPercentage))
				chart.Size = New NSizeL(New NLength(60, NRelativeUnit.ParentPercentage), New NLength(73, NRelativeUnit.ParentPercentage))

				' add a bar series
				Dim bar As NBarSeries = CType(chart.Series.Add(SeriesType.Bar), NBarSeries)
				bar.Name = "Simple Bar Chart"
				bar.BarShape = BarShape.SmoothEdgeBar
				bar.Legend.Mode = SeriesLegendMode.DataPoints
				bar.DataLabelStyle.Format = "<value> <label>"

				bar.AddDataPoint(New NDataPoint(16, "Agriculture"))
				bar.AddDataPoint(New NDataPoint(42, "Construction"))
				bar.AddDataPoint(New NDataPoint(56, "Manufacturing"))
				bar.AddDataPoint(New NDataPoint(23, "Services"))
				bar.AddDataPoint(New NDataPoint(47, "Healthcare"))
				bar.AddDataPoint(New NDataPoint(38, "Finance"))

				' apply style sheet
				Dim styleSheet As NStyleSheet = NStyleSheet.CreatePredefinedStyleSheet(PredefinedStyleSheet.BrightMultiColor)
				styleSheet.Apply(nChartControl1.Document)
			End If
		End Sub
	End Class
End Namespace

File: Examples/Chart/WebForm/CSharp/Examples/WebControl/Printing/NPrintingUC.ascx.cs
using System;
using System.Data;
using System.Drawing;
using System.Web;
using System.Web.UI.WebControls;
using System.Web.UI.HtmlControls;
using System.Web.UI;

using Nevron.GraphicsCore;
using Nevron.Chart;
using Nevron.Chart.WebForm;

namespace Nevron.Examples.Chart.WebForm
{
	public partial class NPrintingUC : NExampleUC
	{
		protected void Page_Load(object sender, EventArgs e)
		{
			nChartControl1.StateManager = new NChartBatonSessionStateManager(Context, "Nevron.Examples.Chart.WebForm.NPrintingUC");
            nChartControl1.BackgroundStyle.FrameStyle.Visible = false;

			if (nChartControl1.RequiresInitialization)
			{
				// enable jittering (full scene antialiasing)
				nChartControl1.Settings.JitterMode = JitterMode.Enabled;

				// set a chart title
				NLabel title = nChartControl1.Labels.AddHeader("Welcome to Nevron Chart for .NET");
				title.TextStyle.BackplaneStyle.Visible = false;
				title.TextStyle.FillStyle = new NColorFillStyle(Color.Navy);
				title.TextStyle.ShadowStyle.Type = ShadowType.GaussianBlur;
				title.ContentAlignment = ContentAlignment.BottomRight;
				title.Location = new NPointL(
					new NLength(2, NRelativeUnit.ParentPercentage),
					new NLength(2, NRelativeUnit.ParentPercentage));

				// setup the legend
				NLegend legend = nChartControl1.Legends[0];
				legend.FillStyle.SetTransparencyPercent(50);
				legend.OuterBottomBorderStyle.Width = new NLength(0, NGraphicsUnit.Pixel);
				legend.OuterLeftBorderStyle.Width = new NLength(0, NGraphicsUnit.Pixel);
				legend.OuterRightBorderStyle.Width = new NLength(0, NGraphicsUnit.Pixel);
				legend.OuterTopBorderStyle.Width = new NLength(0, NGraphicsUnit.Pixel);
				legend.HorizontalBorderStyle.Width = new NLength(0, NGraphicsUnit.Pixel);
				legend.VerticalBorderStyle.Width = new NLength(0, NGraphicsUnit.Pixel);

				// setup the chart
				NChart chart = nChartControl1.Charts[0];
				chart.Enable3D = true;
				chart.Axis(StandardAxis.Depth).Visible = false;
				chart.LightModel.SetPredefinedLightModel(PredefinedLightModel.GlitterLeft);
				chart.Projection.SetPredefinedProjection(PredefinedProjection.OrthogonalHalf);
				chart.BoundsMode = BoundsMode.Stretch;
				chart.Location = new NPointL(
					new NLength(10, NRelativeUnit.ParentPercentage),
					new NLength(17, NRelativeUnit.ParentPercentage));
				chart.Size = new NSizeL(
					new NLength(60, NRelativeUnit.ParentPercentage),
					new NLength(73, NRelativeUnit.ParentPercentage));

				// add a bar series
				NBarSeries bar = (NBarSeries)chart.Series.Add(SeriesType.Bar);
				bar.Name = "Simple Bar Chart";
				bar.BarShape = BarShape.SmoothEdgeBar;
				bar.Legend.Mode = SeriesLegendMode.DataPoints;
				bar.DataLabelStyle.Format = "<value> <label>";

				bar.AddDataPoint(new NDataPoint(16, "Agriculture"));
				bar.AddDataPoint(new NDataPoint(42, "Construction"));
				bar.AddDataPoint(new NDataPoint(56, "Manufacturing"));
				bar.AddDataPoint(new NDataPoint(23, "Services"));
				bar.AddDataPoint(new NDataPoint(47, "Healthcare"));
				bar.AddDataPoint(new NDataPoint(38, "Finance"));

				// apply style sheet
				NStyleSheet styleSheet = NStyleSheet.CreatePredefinedStyleSheet(PredefinedStyleSheet.BrightMultiColor);
				styleSheet.Apply(nChartControl1.Document);
			}
		}
	}
}

